### PR TITLE
[DOCS] Document data stream stats API

### PR DESCRIPTION
**Pull request by jrodewig**
Documents the data streams stats API added with #58707

### CODE REVIEW

**Comment by jrodewig** [comment]
CI expected to fail until #58707 is merged.

**Comment by jrodewig** [comment]
@elasticmachine update branch

File: docs/reference/indices/data-stream-stats.asciidoc [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+(Optional, string)
+Comma-separated list of data streams used to limit the request. Wildcard
+expressions (`*`) are supported.
++
+To target all data streams in a cluster, omit this parameter or use `*`.
+
+
+[[data-stream-stats-api-query-params]]
+==== {api-query-parms-title}
+
```
Comment (jrodewig) [inline docs/reference/indices/data-stream-stats.asciidoc:77]: I intentionally left out `expand_wildcards` as we plan to remove that param later.

File: docs/reference/indices/data-stream-stats.asciidoc [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+Current number of backing indices for the data stream.
+
+`store_size_bytes`::
+(integer)
+Total size, in bytes, of all shards for the data stream's backing indices.
+
+`maximum_timestamp`::
+(integer)
+The data stream's highest `@timestamp` value, converted to milliseconds since
+the https://en.wikipedia.org/wiki/Unix_time[Unix epoch].
```
Comment (jbaiera) [inline docs/reference/indices/data-stream-stats.asciidoc:133]: As Lee mentioned in https://github.com/elastic/elasticsearch/pull/59575#discussion_r454671627, we should probably note that this timestamp value is best effort.

Comment (jrodewig) [reply docs/reference/indices/data-stream-stats.asciidoc:133]: Thanks. Added a note with 099992fc224.

File: docs/reference/indices/data-stream-stats.asciidoc [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+====
+`data_stream`::
+(string)
+Name of the data stream.
+
+`backing_indices`::
+(integer)
+Current number of backing indices for the data stream.
+
+`store_size_bytes`::
```
Comment (jbaiera) [inline docs/reference/indices/data-stream-stats.asciidoc:126]: we also have human readable versions of this field and `total_store_size`, if we wanted to document those.

**Review by jbaiera — APPROVED** [review]
LGTM! Thanks James!

**Comment by jrodewig** [comment]
Thanks @jbaiera!

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Comment by jrodewig** [comment]
I believe https://github.com/elastic/elasticsearch/issues/59802 will break the current doc snippets.

**Comment by jrodewig** [comment]
@elasticmachine update branch

**Pull request merged**